docs/book/src/clusterctl/commands/upgrade.md

```diff
@@ -71,3 +83,18 @@ components YAML/at the installation time.
 User is required to re-apply flag values after the upgrade completes.
 
 </aside>
+
+<aside class="note">
+
+<h1> Upgrading to pre-release provider versions </h1>
+
+In order to upgrade to a provider's pre-release version, we can do the
+following:
+
+```shell
+clusterctl upgrade apply \
+  --management-group capi-system/cluster-api  \
+  --infrastructure capv-system/vsphere:v0.7.0-alpha.0
```

Comment:
In order to be equivalent to `clusterctl upgrade apply  --contract v1alpha3` it is required to upgrade all the providers in the management group, including core provider, bootstrap provider and control-plane provider.
Is it possible to make this explicit in the example or to document this?

Reply:
I added a brief description above on line 59.

Reply:
The description at line 59 is ok, but IMO it is difficult to relate this with this example. What about changing this example into:

```shell
clusterctl upgrade apply --management-group capi-system/cluster-api \
    --core capi-system/cluster-api:v0.3.1 \
    --bootstrap capi-kubeadm-bootstrap-system/kubeadm:v0.3.1 \
    --control-plane capi-kubeadm-control-plane-system/kubeadm:v0.3.1 
  --infrastructure capv-system/vsphere:v0.7.0-alpha.0
```
And add a sentence like e.g.
Please note that in this case, all the providers/version should be listed explicitly on the command.